src/Microsoft.ML.Api/DataViewConstructionUtils.cs

```diff
@@ -400,16 +396,37 @@ protected DataViewBase(IHostEnvironment env, string name, InternalSchemaDefiniti
 
             public abstract long? GetRowCount();
 
-            public abstract IRowCursor GetRowCursor(Func<int, bool> predicate, Random rand = null);
+            public abstract RowCursor GetRowCursor(Func<int, bool> predicate, Random rand = null);
 
-            public IRowCursor[] GetRowCursorSet(out IRowCursorConsolidator consolidator, Func<int, bool> predicate,
+            public RowCursor[] GetRowCursorSet(out IRowCursorConsolidator consolidator, Func<int, bool> predicate,
                 int n, Random rand = null)
             {
                 consolidator = null;
                 return new[] { GetRowCursor(predicate, rand) };
             }
 
-            public abstract class DataViewCursorBase : InputRowBase<TRow>, IRowCursor
+            public sealed class WrappedCursor : RowCursor
```

Comment:
>public [](start = 12, length = 6)

can this be internal?

Reply:
Hello @Ivanidzo4ka , it is, as it is a nested type inside the `internal static class DataViewConstructionUtils`.

Reply:
Yeah, you are right, it's a nested class in public class which is also nested class in internal class.
Or how cool kids says this days: Yo dog, I heard you like nested classes, so i put nested class in your nested class so you can hide all your stuff in nested classes.

---
In reply to: [238777088](https://github.com/dotnet/machinelearning/pull/1814#discussion_r238777088) [](ancestors = 238777088)